libraries/tracker-core/src/payload.ts

```diff
@@ -151,16 +168,61 @@ export function payloadBuilder(): PayloadBuilder {
  * @returns The request builder, with add and build methods
  */
 export function payloadJsonProcessor(encodeBase64: boolean): JsonProcessor {
-  return (payloadBuilder: PayloadBuilder, jsonForProcessing: EventJson) => {
-    for (const json of jsonForProcessing) {
-      const str = JSON.stringify(json.json);
+  return (
```

Comment:
I'm also worried about the added complexity especially since the `addJson` function can be used by external plugins and who knows how they use it...

I tried to clean up the code a bit and also remove the `any` usage where possible. Please take another look when you get a chance.